### DISCUSSION TITLE
media-library:clean does not seem to clean responsive images when withResponsiveImages() has been removed from registerMediaConversions()

### DISCUSSION THREAD

**Discussion by jaap** [spatie/laravel-medialibrary#2397] [Q&A]
We where using responsive images for a while but implemented a different way lately.

In the models where we were using the responsive images we removed the `withResponsiveImages()`

So, from

```php
    public function registerMediaConversions(Media $media = null): void
    {
        $this->addMediaConversion('medium')
            ->width(500)
            ->height(760)
            ->withResponsiveImages();
    }
```
to
```php
    public function registerMediaConversions(Media $media = null): void
    {
        $this->addMediaConversion('medium')
            ->width(500)
            ->height(760);
    }
```

After that I run `media-library:clean` all the files are still there and the svg's are still present in the database

Running `media-library:clean --dry-run` only shows me some orphans and a couple of unused conversions, but not the responsive ones.

I tried debugging it directly in the `CleanCommand` class but can't really figure out what it's actually doing at  `deleteResponsiveImagesForDeprecatedConversions()`

I added some comments what I found in this function

```php
    protected function deleteResponsiveImagesForDeprecatedConversions(Media $media): void
    {
        $conversionNames = ConversionCollection::createForMedia($media)
            ->map(fn (Conversion $conversion) => $conversion->getName())
            ->push('media_library_original');

        $responsiveImagesGeneratedFor = array_keys($media->responsive_images); // Expected conversion name 'medium' is here.

        collect($responsiveImagesGeneratedFor)
            ->map(fn (string $generatedFor) => $media->responsiveImages($generatedFor))
            ->reject(fn (RegisteredResponsiveImages $responsiveImages) => $conversionNames->contains($responsiveImages->generatedFor)) /// returns true, also tried hard-coding true/false
            ->each(function (RegisteredResponsiveImages $responsiveImages) {
                // Tried logging here, it does end up at this point
                if (! $this->isDryRun) {
                    $responsiveImages->delete();
                }
            });
    }
```

I tried adding my case to the `CleanConversionsTest` in the hopes of being able to reproduce it in a test, however, it's a bit daunting to me and could not figure it out. Happy to try some things on my local machine to see if I can create a reproducible test, or, even better, hear what I'm doing wrong.

**Comment by michaellindahl** [comment]
So the problem here is that `deleteResponsiveImagesForDeprecatedConversions` is only looking at `deprecatedConversions` not conversions that are still used but no longer using responsive images. I've opened a PR to address this: #3003
